sdjournal/journal_test.go

```diff
@@ -343,3 +344,54 @@ func TestJournalReaderSmallReadBuffer(t *testing.T) {
 		t.Fatalf("Got unexpected message %s", got[1])
 	}
 }
+
+func TestJournalGetUniqueValues(t *testing.T) {
+	j, err := NewJournal()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer j.Close()
+
+	uniqueString := generateRandomField(20)
+	testEntries := []string{"A", "B", "C", "D"}
+	for _, v := range testEntries {
+		err := journal.Send("TEST: "+uniqueString, journal.PriInfo, map[string]string{uniqueString: v})
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(time.Millisecond * 250)
```

Comment:
unfortunately `waitAndNext` is racy when we send a several messages in a loop. Sometimes the test passes and sometime does not. I pushed the implementation that uses `golang.org/x/net/context` since you guys are using go 1.6 for testing but it seems to fail. Starting with go 1.7 `context` is a standard library package. https://golang.org/doc/go1.7#context

Do you want to adjust your tests and add `golang.org/x/net/context` for go 1.6 ? Or do you want me to implement it in a different way?